custom_components/google_home/api.py

```diff
@@ -210,6 +211,17 @@ async def get_alarms_and_timers(
 
         return device
 
+    async def collect_data_from_endpoints(
+        self, device: GoogleHomeDevice, ip_address: str, auth_token: str
+    ) -> GoogleHomeDevice:
+        """Collect data from different endpoints."""
+
+        device = await self.update_alarms_and_timers(device, ip_address, auth_token)
+
+        device = await self.get_or_set_do_not_disturb(device)
```

Comment:
Can these calls be batched into `gather` to be performed in parallel, @KapJI?
That can be a good starting point where we could add more and more parallel calls as we go (hopefully Google Home will open up some more endpoints, like reminders(been requested feature from 2016 :( ))

Reply:
Oh, just saw that `collect_data_from_endpoints` itself is under `gather`, so maybe my thought would not bring much value :)

Reply:
We can, but I'm afraid that querying the same device in parallel may be troublesome. But we can try.